pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -395,6 +399,31 @@ protected void finishOperation() {
     }
   }
 
+  /**
+   * When TTL is enabled for upsert, this function is used to remove expired keys from the primary key indexes.
+   */
+  @Override
+  public void removeExpiredPrimaryKeys(Comparable timestamp) {
+    if (_upsertTTLConfig != null && _upsertTTLConfig.getTtlInMs() > 0) {
```

Comment:
This check is redundant because it can only be invoked when TTL is configured. We may remove the default implementation. Same for `persistSnapshotForStableSegment()`

Reply:
@deemoliu can you address this comment?